v0/destinations/mautic/utils.js

```diff
@@ -0,0 +1,298 @@
+function validatePhone(inputText) {
+  const phoneno = /^\d{10}$/;
+  return phoneno.test(inputText);
+}
+
+//Refines the payload by validating and adding some payload Fields
+const refinePayloadFields = (payload, message) => {
+  const { traits, context } = message;
+  if (
+    (traits && traits?.hasPurchased) ||
+    (context.traits && context.traits?.hasPurchased)
+  ) {
+    let purchased_status = traits?.hasPurchased || context.traits?.hasPurchased;
+    purchased_status = purchased_status.toLowerCase();
+    if (purchased_status === "yes" || purchased_status === "no") {
+      set(payload, "haspurchased", purchased_status);
+    } else {
+      throw new CustomError("Invalid Purchase Status.", 400);
+    }
+  }
+  if ((traits && traits?.role) || (context && context.traits?.role)) {
+    const Role =
+      traits && traits?.role
+        ? message.traits?.role
+        : message.context.traits?.role;
+    if (!ALLOWED_ROLE_VALUES.includes(Role)) {
+      throw new CustomError("This Role is not supported.", 400);
+    }
+    set(payload, "role", Role);
+  }
+  if (
+    (traits && traits?.subscriptionStatus) ||
+    (context.traits && context.traits?.subscriptionStatus)
+  ) {
+    const status =
+      traits && traits?.subscriptionStatus
+        ? message.traits?.subscriptionStatus
+        : message.context.traits?.subscriptionStatus;
+    if (!ALLOWED_SUBSCRIPTION_STATUS.includes(status)) {
+      throw new CustomError("Invalid subscriptionStatus Value.", 400);
+    }
+    set(payload, "subscription_status", status);
+  }
+  if (
+    (traits && traits?.prospectOrCustomer) ||
+    context.traits?.prospectOrCustomer
+  ) {
+    const POC =
+      message.traits?.prospectOrCustomer ||
+      message.context.traits?.prospectOrCustomer;
+    if (!ALLOWED_POC.includes(POC)) {
+      throw new CustomError(
+        "prospectOrCustomer can only be either prospect or customer or null.",
+        400
+      );
+    }
+    set(payload, "prospect_or_customer", POC);
+  }
+  return payload;
+};
+
+//Constructs the address1 and address2 field
+//if address is given as string or object
+const deduceAddressFields = message => {
+  const { traits, context } = message;
+  let address1;
+  let address2;
+  if (
+    (traits !== undefined && traits?.address) ||
+    (context.traits && context.traits?.address)
+  ) {
+    let add =
+      traits && traits?.address
+        ? message.traits.address
+        : message.context.traits.address;
+    if (typeof add === "object") {
+      add = Object.keys(add).reduce(function(res, v) {
+        return res.concat(add[v], " ");
+      }, "");
+    }
+    const validLengthAddress = add.length > 128 ? add.substring(0, 127) : add;
+    address1 = validLengthAddress.substring(0, 63);
+    address2 = validLengthAddress.substring(64, validLengthAddress.length);
+  }
+  return { address1, address2 };
+};
+
+//Validates the generated payload fro specific fields
+const validatePayload = payload => {
```

Comment:
Please use javascript function description standard for all the util functions. See `searchContactId` for reference. Also you can have a look at https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/#functions